text/0052-tx-summary-digest.md

```diff
@@ -0,0 +1,510 @@
+  probability.
+* In the final case, the `TxOutUnblindingData` doesn't contain an address or `tx_private_key`. This can occur normally when the `TxOut`
+  was not created using `TransactionBuilder.add_output` interface, but rather
+  came as a required (or partial fill) output from a Signed Contingent Input which was added to the `Tx`.
+  In this case it is not addressed to ourself, and due to the anonyminity properties of the MCIP 31 atomic swaps, we cannot know who
+  it is actually addressed to. We can only know that it goes to our anonymous swap counterparty.
+  However, it's not okay to say that addresses / tx_private keys are optional for things that don't match to ourselves, because this
+  allows the (adversarial) host computer to simply omit the address when they don't want to tell the hardware wallet where a payment is going.
+  To prevent this tampering, `TxOutSummary` has a flag `associated_to_input_rules`, and this is in the `TxSummary`, checked also by the enclave
+  when it derives `TxSummary` from `TxPrefix`. If this flag is false, then the hardware device knows that the host computer has lied to it, by
+  witholding the address information and trying to pretend this is an SCI output.
+  Therefore the decision tree for this third and final case is simply, check if the `TxOutSummary.associated_input_rules` flags is true, and if
+  so, attribute it to the swap counterparty. Otherwise, signal an error, that the host computer did not provide enough information for validation
+  to proceed.
+
+For inputs, we can use the following decision tree:
+* Given a `TxInSummary`, the `TxSummaryUnblindingData` contains a corresponding `UnmaskedAmount`.
+  This include the value, token id, and blinding factor corresponding to the pseudo-output commitment.
+  We can compute the corresponding Pedersen commitment to these deterministically, and check if this matches
+  the pseudo-output commitment in the `TxInSummary`. This came from the `TxSummary` which is checked by the enclave,
+  and so we know that it must be accurate. If we manage to unblind it, then we know this is the correct unblinding,
+  because it is discrete-log hard to find another unmasking of the Pedersen commitment. If we can't unblind it, then
+  the hardware device knows that the computer has lied / provided a false unmasking, but there is no legitimate reason
+  for the computer to fail to do this, because regardless of whether this is our own input or a signed input from a
+  counterparty, we can expect to have the unmasked amount or there is no way we can possibly spend it.
+* Now having the amount in hand, the next step is to check the `has_input_rules` flag from `TxInSummary`. We know that
+  this flag is accurate because the consensus enclave checks the `TxSummary` and can see which inputs actually have rules.
+  If there are no rules, then we know that we must be signing for this transaction. If there are rules then we attribute
```

Comment:
signing for this transaction or for this input?

Reply:
thanks